cub/cub/util_macro.cuh

```diff
@@ -137,6 +137,9 @@ _CCCL_DIAG_SUPPRESS_CLANG("-Wattributes")
 #if !defined(_CCCL_CUDA_COMPILER_NVHPC)
 _CCCL_DIAG_SUPPRESS_NVHPC(attribute_requires_external_linkage)
 #endif // !_CCCL_CUDA_COMPILER_NVHPC
+#if defined(__ICC)
```

Comment:
Important: this is technically not the right define, we should use our compiler detection instead
```suggestion
#if defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)
```

Reply:
```
cat main.cu 
#include <cuda/__cccl_config>
#include <iostream>

int main() {
#if defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)
	std::cout << "defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)" << std::endl;
#endif

#if defined(__ICC)
	std::cout << "defined(__ICC)" << std::endl;
#endif
}


/usr/local/cuda/bin/nvcc -ccbin=icc main.cu -I/cccl/libcudacxx/include


./a.out 
defined(__ICC)
```